src/AspNet.Security.OAuth.Introspection/Events/ValidateAudienceContext.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AspNet.Security.OAuth.Introspection {
+    public class ValidateAudienceContext : BaseIntrospectionContext {
+        public ValidateAudienceContext(
+            HttpContext context, 
+            OAuthIntrospectionOptions options,
+            JToken payload) 
+            : base(context, options) {
+            Payload = payload;
+            var audiences = payload[OAuthIntrospectionConstants.Claims.Audience];
```

Comment:
Too much in the ctor? Or the functionality of the `Validate` methods themselves?